### PR TITLE
Code clean up

### PR DESCRIPTION
**Pull request by Manpreet-Bhatti**
Following functions have typescript errors and need revision:

- [x] formatLanguages
- [x] generateMarkDownTree
- [x] getCopyToClipboardContents
- [x] deepCopyFunction
- [x] filterChange

### CODE REVIEW

File: src/components/MarkdownDisplay.tsx [hunk 1]
```diff
@@ -1,41 +1,29 @@
...
     // Update the document title using the browser API
-    setClipboardContent(
-      getCopyToClipboardContents(treeCore, filtering(filter))
-    );
+    setClipboardContent(getCopyToClipboardContents(treeCore, filter));
   }, [filter]);
 
   const [clipboardContent, setClipboardContent] = useState<string[]>(
-    getCopyToClipboardContents(treeCore, filtering(filter))
+    getCopyToClipboardContents(treeCore, filter),
```
Comment (ralph-dev) [inline src/components/MarkdownDisplay.tsx:26]:
```suggestion
    getCopyToClipboardContents(treeCore, filter)
```

File: src/utils/deleteFileFromPath/deleteFileFromPath.ts [hunk 2]
```diff
@@ -1,50 +1,59 @@
...
     }
     if (
-      treeCore[index].path === path &&
-      getFileTypeFromPath(path) === FileType.FOLDER
+      treeCore[index].path === path
+      && getFileTypeFromPath(path) === FileType.FOLDER
     ) {
-      lastDeletedOrder += 1;
-      treeCore[index].deletedOrder = lastDeletedOrder;
+      treeCore[index].deletedOrder = lastDeletedOrder + 1;
```
Comment (ralph-dev) [inline src/utils/deleteFileFromPath/deleteFileFromPath.ts:51]: 1 is a Magic number, extract to const DELETION_INCREMENT = 1;

**Comment by Manpreet-Bhatti** [comment]
@ralph-dev something wrong with the deepCopy function, I somehow messed things up in getHyperLinkFromPath

File: src/utils/undoDeletions/undoDeletions.ts [hunk 3]
```diff
@@ -12,15 +12,15 @@ import { getFileTypeFromPath } from '../getFileTypeFromPath/getFileTypeFromPath'
 */
 
 let highestDeletedOrder = -1;
+const resetDeletedOrder = -1;
```
Comment (ralph-dev) [inline src/utils/undoDeletions/undoDeletions.ts:15]:
```suggestion
const RESET_DELETED_ORDER = -1;
```
Make sure all caps for CONSTANTS :)

**Comment by Manpreet-Bhatti** [comment]
@ralph-dev 

Realized the TypeError screenshot didn't get posted since it was on my localhost -- my bad. Please refer to the image below for the TypeError.

Just a thought: should we revert back to the old deepCopyFunction with the TS errors fixed since it worked? The only thing that was prevalent was the ESLint error -- fixing that basically messed up the rest of the project.

image

**Pull request merged**